myriad-scheduler/src/main/java/com/ebay/myriad/state/SchedulerState.java

```diff
@@ -198,7 +225,54 @@ public MyriadState getMyriadState() {
         return taskStatuses;
     }
 
-    public boolean hasTask(Protos.TaskID taskID) {
+    public synchronized boolean hasTask(Protos.TaskID taskID) {
         return this.tasks.containsKey(taskID);
     }
-}
\ No newline at end of file
+
+    public synchronized Protos.FrameworkID getFrameworkID() {
+         return this.frameworkId;
+    }
+
+    public synchronized void setFrameworkId(Protos.FrameworkID newFrameworkId) {
+        this.frameworkId = newFrameworkId;
+        updateStateStore();
+    }
+
+    private synchronized void updateStateStore() {
+        if (this.stateStore == null) {
+            LOGGER.debug("Could not update state to state store as HA is disabled");
+            return;
+        }
+
+        try {
+            StoreContext sc = new StoreContext(frameworkId, tasks, pendingTasks,
+                stagingTasks, activeTasks, lostTasks, killableTasks);
+            stateStore.storeMyriadState(sc);
+            LOGGER.info("Scheduler state updated to state store");
+        } catch (Exception e) {
+            LOGGER.error("Failed to update scheduler state to state store", e);
+        }
+    }
+
+    private synchronized void loadStateStore() {
+        if (this.stateStore == null) {
+            LOGGER.debug("Could not load state from state store as HA is disabled");
+            return;
+        }
+
+        try {
+            StoreContext sc = stateStore.loadMyriadState();
+            if (sc != null) {
+                this.frameworkId = sc.getFrameworkId();
+                this.tasks.putAll(sc.getTasks());
+                this.pendingTasks.addAll(sc.getPendingTasks());
+                this.stagingTasks.addAll(sc.getStagingTasks());
+                this.activeTasks.addAll(sc.getActiveTasks());
+                this.lostTasks.addAll(sc.getLostTasks());
+                this.killableTasks.addAll(sc.getKillableTasks());
+            }
```

Comment:
I suggest we add a .info that tells that we've successfully loaded the state store.

Also, a .debug that prints some metadata like framework id, number of pending/staging/active/lost/killable tasks recovered will be useful.